src/models/UserEmail.ts

```diff
@@ -75,6 +78,33 @@ export class UserEmail extends MySqlModel {
     return null;
   }
 
+  // Custom Insert logic
+  static async insert(
+    apolloContext: MyContext,
+    table: string,
```

Comment:
Hi @andrewebdev and @briri. I have a suggestion.
Can we just add an `id` property to the `UserEmail` class, and set it to `options.userId`, and then in the MySqlModel, we could update the `insert` function to fallback to `obj.id` when there is no token: 
```
  static async insert(
    apolloContext: MyContext,
    table: string,
    obj: MySqlModel,
    reference = 'undefined caller',
    skipKeys?: string[]
  ): Promise<number> {
    const id = obj.id;
    // Update the creator/modifier info
    const currentDate = getCurrentDate();
    obj.createdById = apolloContext?.token?.id | Number(id);
    obj.created = currentDate;
    obj.modifiedById = apolloContext?.token?.id | Number(id);
    obj.modified = currentDate;
```

Reply:
The reason I added an `id` property, is that `MySqlModel` only as an `id` property for the `user id`. We could also update `UserEmail` to use `id` instead of `userId`?

I agree with Brian that it doesn't make sense to create another `insert` function to override the existing one in `MySqlModel`.

Reply:
great idea Juliet!

Reply:
@andrewebdev I actually updated my suggestion above. We can't add an `id` property to `UserEmail`, because it is reserved as a unique identifier for `UserEmail`. Instead, we should extend the definition of `obj` in `insert` function to accept `userId`.

Reply:
@jupiter007 Wouldn't it be better to provide a single parameter in the insert() call to add an optional user override?

I feel that, extending the obj to include user is making too many assumptions of what the object might look like in future. For example, a future object might have it's own userId field, unrelated to the owner of the object.

I think an optional parameter at the end of the function to pass in a user is more explicit. In the function, if we see we have this parameter set, then we use that instead of what's on the token; Otherwise we just use the token user id as we currently do.

Reply:
@andrewebdev yea the reason I didn't suggest a new parameter was because it would need to be passed around everywhere, and it wouldn't apply to any other class, since I believe they will all have a token in their context.